tests/test_tarfile.py

```diff
@@ -0,0 +1,47 @@
+import tarfile
+
+with tarfile.open("test.tar.xz", "w:xz") as tar:
+    pass
+
+# Test with valid preset values
+for preset in range(10):
+    with tarfile.open("test.tar.xz", "w:xz", preset=preset) as tar:
+        pass
+
+# Test with invalid preset values
+try:
+    with tarfile.open("test.tar.xz", "w:xz", preset=-1) as tar:
+        pass
+except ValueError:
+    pass
+
+try:
+    with tarfile.open("test.tar.xz", "w:xz", preset=10) as tar:
+        pass
+except ValueError:
+    pass
```

Comment:
Similarly, this won't do anything in the tests. We'd need to check it like this:

```suggestion
tarfile.open("test.tar.xz", "w:xz", preset=-1)  # type: ignore
tarfile.open("test.tar.xz", "w:xz", preset=10)  # type: ignore
```

Reply:
added the suggested changes accordingly